avocado/plugins/run.py

```diff
@@ -151,17 +151,21 @@ def configure(self, parser):
                          "final multiplex tree.")
 
     def _activate(self, args):
-        # Extend default multiplex tree of --mux_inject values
+        # Merge the multiplex_files
+        multiplex_files = getattr(args, "multiplex_files", None)
+        if multiplex_files:
+            args.mux.data_merge(multiplexer.yaml2tree(multiplex_files))
+
+        # Extend default multiplex tree of --mux-inject values
         for value in getattr(args, "mux_inject", []):
-            value = value.split(':', 2)
+            value = value.split(':', 3)
```

Comment:
It's unfortunate that this same code is found in both `avocado/plugins/run.py` and `avocado/plugins/multiplex.py`.  Let's try to put it in a common location.